src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1476,7 +1477,9 @@ public void addVisitingPlayer(final EntityPlayer player)
         {
             visitingPlayers.add(player);
             LanguageHandler.sendPlayerMessage(player, ENTERING_COLONY_MESSAGE, this.getPermissions().getOwnerName());
-            LanguageHandler.sendPlayersMessage(getImportantMessageEntityPlayers(), ENTERING_COLONY_MESSAGE_NOTIFY, player.getName(), this.getName());
+            if(!PermissionAPI.hasPermission(player, "minecolonies.colony.visitor.silent")) {
```

Comment:
make this an or `if the player is OP`

Reply:
Added the permission by default to operator in MineColonies.java. Seemed cleaner then directly checking for operator status here (only way I know to check it is to grab the operator list and see if UUID is in it).

This does have the slight disadvantage though for users of default Forge perm handler in single player that Forge considers everyone operator if they are allowed to run all commands (See DefaultPermissionsHandler.java in Forge).